R/navs-legacy.R

```diff
@@ -105,36 +105,220 @@ navset_hidden <- function(..., id = NULL, selected = NULL,
-                     collapsible = TRUE, fluid = TRUE) {
+navset_bar <- function(
+  ...,
+  title = NULL,
+  id = NULL,
+  selected = NULL,
+  sidebar = NULL,
+  fillable = TRUE,
+  gap = NULL,
+  padding = NULL,
+  header = NULL,
+  footer = NULL,
+  fluid = TRUE,
+  navbar_options = NULL,
+  position = deprecated(),
+  bg = deprecated(),
+  inverse = deprecated(),
+  collapsible = deprecated()
+) {
   padding <- validateCssPadding(padding)
   gap <- validateCssUnit(gap)
 
-  navs_bar_(
-    ..., title = title, id = id, selected = selected,
-    sidebar = sidebar, fillable = fillable,
-    gap = gap, padding = padding,
+  .navbar_options <- navbar_options_resolve_deprecated(
+    options_user = navbar_options,
     position = position,
-    header = header, footer = footer,
-    bg = bg, inverse = inverse,
-    collapsible = collapsible, fluid = fluid,
+    bg = bg,
+    inverse = inverse,
+    collapsible = collapsible
+  )
+
+  navs_bar_(
+    ...,
+    title = title,
+    id = id,
+    selected = selected,
+    sidebar = sidebar,
+    fillable = fillable,
+    gap = gap,
+    padding = padding,
+    header = header,
+    footer = footer,
+    fluid = fluid,
+    position = .navbar_options$position,
+    bg = .navbar_options$bg,
+    inverse = .navbar_options$inverse,
+    collapsible = .navbar_options$collapsible,
+    underline = .navbar_options$underline,
     # theme is only used to determine whether legacy style markup should be used
     # (and, at least at the moment, we don't need legacy markup for this exported function)
     theme = bs_theme()
   )
 }
 
+#' Create a set of navbar options
+#' 
+#' A `navbar_options()` object captures options specific to the appearance and
+#' behavior of the navbar, independent from the content displayed on the page.
+#' This helper should be used to create the list of options expected by
+#' `navbar_options` in [page_navbar()] and [navset_bar()].
+#' 
+#' ## Changelog
+#' 
+#' This function was introduced in \pkg{bslib} v0.9.0, replacing the `position`,
+#' `bg`, `inverse`, `collapsible` and `underline` arguments of [page_navbar()]
+#' and [navset_bar()]. Those arguments are deprecated with a warning and will be
+#' removed in a future version of \pkg{bslib}.
+#' 
+#' @examples
+#' navbar_options(position = "static-top", bg = "#2e9f7d", underline = FALSE)
+#' 
+#' @inheritParams shiny::navbarPage
+#' @param bg a CSS color to use for the navbar's background color.
+#' @param inverse Either `TRUE` for a light text color or `FALSE` for a dark
+#'   text color. If `"auto"` (the default), the best contrast to `bg` is chosen.
+#' @param underline Whether or not to add underline styling to page or navbar
+#'   links when active or focused.
+#' 
+#' @returns Returns a list of navbar options.
+#' 
+#' @export
+navbar_options <- function(
+  position = c("static-top", "fixed-top", "fixed-bottom"),
+  bg = NULL,
+  inverse = "auto",
+  collapsible = TRUE,
+  underline = TRUE
+) {
```

Comment:
Should we include `...` here to make future expansion easier? Maybe even making all arguments required?

Reply:
I made this change. We'll need the cover from `...` and because there's no clear ordering I'd rather not encourage the positional argument usage that would make future expansion difficult.